src/dodal/beamlines/i18.py

```diff
@@ -0,0 +1,286 @@
+        fake=fake_with_ophyd_sim,
+    )
+
+
+crystal_1_metadata = CrystalMetadata(
+    usage="Bragg",
+    type="silicon",
+    reflection=(1, 1, 1),
+    d_spacing=(3.13475, "nm"),
+)
+
+crystal_2_metadata = CrystalMetadata(
+    usage="Bragg",
+    type="silicon",
+    reflection=(1, 1, 1),
+    d_spacing=(3.13475, "nm"),
+)
+
+
+def dcm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> DoubleCrystalMonochromator:
+    return device_instantiation(
+        DoubleCrystalMonochromator,
+        "dcm",
+        f"{BeamlinePrefix(BL).beamline_prefix}-MO-DCM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        bl_prefix=False,
+        temperature_prefix=f"{BeamlinePrefix(BL).beamline_prefix}-DI-DCM-01:",
+        crystal_1_metadata=crystal_1_metadata,
+        crystal_2_metadata=crystal_2_metadata,
+    )
+
+
+@skip_device()
+def i0(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> TetrammDetector:
+    return device_instantiation(
+        TetrammDetector,
+        "i0",
+        "-DI-XBPM-02:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        type="Cividec Diamond XBPM",
+        path_provider=get_path_provider(),
+    )
+
+
+@skip_device()
+def it(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> TetrammDetector:
+    return device_instantiation(
+        TetrammDetector,
+        "it",
+        "-DI-XBPM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+        type="Tetramm",
+        path_provider=get_path_provider(),
+    )
+
+
+@skip_device
+def vfm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> KBMirror:
+    return device_instantiation(
+        KBMirror,
+        "vfm",
+        "-OP-VFM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def hfm(
+    wait_for_connection: bool = True,
+    fake_with_ophyd_sim: bool = False,
+) -> KBMirror:
+    return device_instantiation(
+        KBMirror,
+        "hfm",
+        "-OP-HFM-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def diode(wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False) -> Diode:
+    return device_instantiation(
+        Diode,
+        "diodad7bdiode",
+        "-DI-PHDGN-07:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def main_table(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> Table:
+    return device_instantiation(
+        Table,
+        "table",
+        "-MO-TABLE-01:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
+
+
+def thor_labs_table(
+    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
+) -> Table:
+    return device_instantiation(
+        Table,
+        "table",
+        "-MO-TABLE-02:",
+        wait_for_connection,
+        fake_with_ophyd_sim,
+    )
```

Comment:
You're defining 2 Table classes, but then re-using the same one for both of these tables?

Reply:
??the PVs are different, this is lab-beamline relationship

Reply:
https://github.com/DiamondLightSource/dodal/blob/cace9c93ac32b2c92e8e515821eebedad13b0000/src/dodal/devices/i18/table.py
https://github.com/DiamondLightSource/dodal/blob/cace9c93ac32b2c92e8e515821eebedad13b0000/src/dodal/devices/i18/thor_labs_stag.py

You are defining 2 classes, then re-using the same class for both instances.

Reply:
ok thanks